bundles/binding/org.openhab.binding.cardio2e/src/main/java/org/openhab/binding/cardio2e/internal/Cardio2eBinding.java

```diff
@@ -0,0 +1,944 @@
+	private BundleContext bundleContext;
+
+	
+	/** 
+	 * the refresh interval which is used to poll values from the Cardio2e
+	 * server (optional, defaults to 60000ms)
+	 */
+	private long refreshInterval = 30000;
+	
+	
+	public Cardio2eBinding() {
+	}
+		
+	
+	/**
+	 * Called by the SCR to activate the component with its configuration read from CAS
+	 * 
+	 * @param bundleContext BundleContext of the Bundle that defines this component
+	 * @param configuration Configuration properties for this component obtained from the ConfigAdmin service
+	 */
+	public void activate(final BundleContext bundleContext, final Map<String, Object> configuration) {
+		this.bundleContext = bundleContext;
+
+		logger.debug("activate");	
+		initializeCardio2eBinding(configuration);
+	}
+	
+	/**
+	 * Called by the SCR when the configuration of a binding has been changed through the ConfigAdmin service.
+	 * @param configuration Updated configuration properties
+	 */
+	public void modified(final Map<String, Object> configuration) {
+		// update the internal configuration accordingly
+		logger.debug("modified");
+		logger.debug("Cardio2e binding will be deactivated and reactivated to apply new configuration...");
+		purgeCardio2eBinding();
+		initializeCardio2eBinding(configuration);
+	}
+	
+	/**
+	 * Called by the SCR to deactivate the component when either the configuration is removed or
+	 * mandatory references are no longer satisfied or the component has simply been stopped.
+	 * @param reason Reason code for the deactivation:<br>
+	 * <ul>
+	 * <li> 0 – Unspecified
+     * <li> 1 – The component was disabled
+     * <li> 2 – A reference became unsatisfied
+     * <li> 3 – A configuration was changed
+     * <li> 4 – A configuration was deleted
+     * <li> 5 – The component was disposed
+     * <li> 6 – The bundle was stopped
+     * </ul>
+	 */
+	
+	public void deactivate(final int reason) {
+		this.bundleContext = null;
+		// deallocate resources here that are no longer needed and 
+		// should be reset when activating this binding again
+		logger.debug("deactivate");
+		purgeCardio2eBinding();
+	}
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected String getName() {
+		return "Cardio2e Refresh Service";
+	}
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected void execute() {
+		// the frequently executed code (polling) goes here ...
+		logger.debug("execute() method is called!");
+		if (loggedIn) {
+			//Cardio 2é date and time cyclic ask code used for progressive update only
+			if ((datetimeMaxOffset >= 0) && (pendingDateTimeUpdate.getDateTime() != null)) {
+            	getCardioDateTime();
+            }
+		}
+	}
+
+	public void initializeCardio2eBinding(final Map<String, Object> configuration) {
+		int minDelayBetweenReceivingAndSending = 0;
+		int minDelayBetweenSendings = 0;
+		// Set default values 
```

Comment:
Default values are set in this method initializeCardio2eBinding() in order to not duplicate code. Variables are declared without values and defaults are setted before getting values from config fle, so if these values ware modified and invoked modified() method, these values would be setted to defaults values again before getting new values from config.
Is this wrong?